gmsad/ldap.py

```diff
@@ -28,18 +28,35 @@ def __init__(self, config: configparser.SectionProxy) -> None:
         else:
             host = get_dc(self.config['gMSA_domain'])
         logging.debug("LDAP Server host to contact is %s", host)
+        
+        auto_bind = self.config.get("ldap_auto_bind", True)
```

Comment:
`mypy` doesn't agree with this:
```
gmsad/ldap.py:32: error: Argument 2 to "get" of "SectionProxy" has incompatible type "bool"; expected "Optional[str]"  [arg-type]
```

`self.config` is a `configparser.SectionProxy`, not a `dict`. It should be `self.config.get("ldap_auto_bind", fallback=True)`. Also, I'm not sure that  `mypy` will be happy with a variable that can be a boolean but is manipulated as a `str` (`.upper()`).

`ldap3` wants `auto_bind` to be a `bool` or a `str`, but the `bool` is converted into a `str` (https://github.com/cannatag/ldap3/blob/dev/ldap3/core/connection.py#L257C1-L262C43
). I think it would be better to always provide a `str`, and default to `ldap3.AUTO_BIND_NO_TLS` (equivalent to `auto_bind=True`).

I suggest to check if `ldap_auto_bind` has been set in config. If so, assume it is a string, eventually add the `AUTO_BIND_` prefix, make sure it is known to `ldap3` and use the result as `auto_bind`. Otherwise, set `auto_bind` to `ldap3.AUTO_BIND_NO_TLS`.

Reply:
> Did you mean ignore invalid ldap_auto_bind values and just write a warning and fallback on NO_TLS instead of raising a ValueError ?

No, we definitely want to raise an error if the user provided value is incorrect! I meant that if the `ldap_auto_bind` parameter is not provided by the user, we default to NO_TLS.